cmd/containerboot: TestContainerBoot flake

https://github.com/tailscale/tailscale/actions/runs/24363666577/job/71156848197?pr=19375
```
    main_test.go:1288: phase 2: want exit code 0, got 1
    main_test.go:1247: containerboot output:
        boot: 2026/04/13 20:41:52 kubeclient: this client is not able to write tailscaled Events to the Pod in which it is running.
        			To help with future debugging you can make it able write Events by giving it get,create,patch permissions for Events in the Pod namespace
        			and setting POD_NAME, POD_UID env vars for the Pod.
        boot: 2026/04/13 20:41:52 No authkey found in state Secret and TS_AUTHKEY not provided, login will be interactive if needed.
        boot: 2026/04/13 20:41:52 Starting tailscaled
        boot: 2026/04/13 20:41:52 Waiting for tailscaled socket at /tmp/TestContainerBootkube_shutdown_during_state_write3865334707/001/tmp/tailscaled.sock
        boot: 2026/04/13 20:41:52 Running healthcheck endpoint at [::]:9002/healthz
Warning: oot: 2026/04/13 20:41:52 [warning] failed to symlink socket: mkdir /var/run/tailscale: permission denied
        	To interact with the Tailscale CLI please use `tailscale --socket="/tmp/TestContainerBootkube_shutdown_during_state_write3865334707/001/tmp/tailscaled.sock"`
        boot: 2026/04/13 20:41:52 Running 'tailscale up'
        boot: 2026/04/13 20:41:52 HTTP server at [::]:9002 closed
        boot: 2026/04/13 20:41:52 Checking for consistent state
        boot: 2026/04/13 20:41:52 Waiting for tailscaled to finish writing state to Secret "tailscale"
        boot: 2026/04/13 20:41:52 Sending SIGTERM to tailscaled
        boot: 2026/04/13 20:41:52 failed to auth tailscale: failed to auth tailscale: tailscale up failed: context canceled
--- PASS: TestContainerBoot/kube_storage_auth_once (0.20s)
=== CONT  TestContainerBoot/clears_reissue_authkey_on_change
=== CONT  TestContainerBoot/auth_key_once_extra_args_override_dns
--- FAIL: TestContainerBoot/kube_shutdown_during_state_write (0.34s)
```

--- a/cmd/containerboot/tailscaled.go
+++ b/cmd/containerboot/tailscaled.go
@@ -150,7 +150,15 @@ func tailscaleUp(ctx context.Context, cfg *settings) error {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("tailscale up failed: %v", err)
+		if ctxErr := ctx.Err(); ctxErr != nil {
+			// A canceled context kills the command, and cmd.Run can
+			// report the subprocess's death ("signal: killed") rather
+			// than the context error that caused it. Return the
+			// context error so that callers (and ultimately main) can
+			// recognize a graceful shutdown with errors.Is.
+			return fmt.Errorf("tailscale up failed: %w", ctxErr)
+		}
+		return fmt.Errorf("tailscale up failed: %w", err)
 	}
 	return nil
 }
@@ -180,7 +188,11 @@ func tailscaleSet(ctx context.Context, cfg *settings) error {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("tailscale set failed: %v", err)
+		if ctxErr := ctx.Err(); ctxErr != nil {
+			// See the equivalent check in tailscaleUp.
+			return fmt.Errorf("tailscale set failed: %w", ctxErr)
+		}
+		return fmt.Errorf("tailscale set failed: %w", err)
 	}
 	return nil
 }

